File: global/global.cs
using System.Collections.Generic;
using System;

// namespace LeetCode
// {
public class ListNode
{
    public int val;
    public ListNode next;
    public ListNode(int val = 0, ListNode next = null)
    {
        this.val = val;
        this.next = next;
    }
}

public class TreeNode
{
    public int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode(int val = 0, TreeNode left = null, TreeNode right = null)
    {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

public class PrintCur
{
    private int Deep(TreeNode root)
    {
        if (root == null)
        {
            return 0;
        }
        int left = Deep(root.left);
        int right = Deep(root.right);
        return Math.Max(left, right) + 1;
    }

    private void Dfs(TreeNode root, List<List<string>> ret, int row, int col, int treeDeep)
    {
        if (root == null)
        {
            return;
        }
        ret[row][col] = root.val.ToString();
        if (root.left != null)
            Dfs(root.left, ret, row + 1, col - (1 << (treeDeep - row - 1)), treeDeep);
        if (root.right != null)
            Dfs(root.right, ret, row + 1, col + (1 << (treeDeep - row - 1)), treeDeep);
    }

    public List<List<string>> PrintTree(TreeNode root)
    {
        int treeDeep = Deep(root) - 1; // 根节点高度为0
        int row = treeDeep + 1;
        int col = (1 << row) - 1;
        List<List<string>> ret = new List<List<string>>();
        for (int i = 0; i < row; i++)
        {
            ret.Add(new List<string>(new string[col]));
        }
        Dfs(root, ret, 0, (col - 1) / 2, treeDeep);
        for (int i = 0; i < ret.Count; i++)
        {
            for (int j = 0; j < ret[i].Count; j++)
            {
                Console.Write(ret[i][j] + " ");
            }
            Console.WriteLine();
        }
        return ret;
    }

    public void PrintListNode(ListNode head)
    {
        while (head != null)
        {
            System.Console.Write(head.val + " ");
            head = head.next;
        }
        System.Console.WriteLine();
    }
    public void PrintArray(int[] arr)
    {
        foreach (var item in arr)
        {
            System.Console.Write(item + " ");
        }
        System.Console.WriteLine();
    }
    public void PrintArray<T>(List<T> arr)
    {
        foreach (var item in arr)
        {
            System.Console.Write(item + " ");
        }
        System.Console.WriteLine();
    }
    public void PrintArray(int[][] arr)
    {
        foreach (var item in arr)
        {
            PrintArray(item);
        }
    }

    public void PrintArray<T>(List<List<T>> arr)
    {
        foreach (var item in arr)
        {
            PrintArray(item);
        }
    }
}
// }

File: global/global.ts
// 定义TreeNode结构
export class TreeNode {
    val: number;
    left: TreeNode | null;
    right: TreeNode | null;

    constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
        this.val = val === undefined ? 0 : val;
        this.left = left === undefined ? null : left;
        this.right = right === undefined ? null : right;
    }
}

// 定义ListNode结构
export class ListNode {
    val: number;
    next: ListNode | null;

    constructor(val?: number, next?: ListNode | null) {
        this.val = val === undefined ? 0 : val;
        this.next = next === undefined ? null : next;
    }
}

// 二叉树打印函数方便测试
export class PrintCur {
    // 计算二叉树深度
    deep(root: TreeNode | null): number {
        if (root === null) {
            return 0;
        }
        const left = this.deep(root.left);
        const right = this.deep(root.right);
        return Math.max(left, right) + 1;
    }

    // 深度优先搜索打印二叉树
    dfs(root: TreeNode | null, ret: string[][], row: number, col: number, deep: number): void {
        if (root === null) {
            return;
        }
        ret[row][col] = root.val.toString();
        if (root.left !== null) {
            this.dfs(root.left, ret, row + 1, col - (1 << (deep - row - 1)), deep);
        }
        if (root.right !== null) {
            this.dfs(root.right, ret, row + 1, col + (1 << (deep - row - 1)), deep);
        }
    }

    // 打印二叉树
    printTree(root: TreeNode | null): string[][] {
        const treeDeep = this.deep(root) - 1; // 根节点高度为0
        const row = treeDeep + 1;
        const col = (1 << row) - 1;
        const ret: string[][] = new Array(row).fill(null).map(() => new Array(col).fill(""));
        this.dfs(root, ret, 0, (col - 1) / 2, treeDeep);
        for (let i = 0; i < ret.length; i++) {
            for (let j = 0; j < ret[i].length; j++) {
                console.log(ret[i][j] + " ", { end: '' });
            }
            console.log("\n", { end: '' });
        }
        return ret;
    }

    // 打印一维数组
    printVecter(buff: number[]): void {
        for (const num of buff) {
            console.log(num + " ", { end: '' });
        }
        console.log("\n", { end: '' });
    }

    // 打印二维数组
    printVecterDouble(buff: number[][]): void {
        for (const num of buff) {
            this.printVecter(num);
        }
    }

    // 打印链表
    printList(head: ListNode | null): void {
        while (head !== null) {
            console.log(head.val + " ", { end: '' });
            head = head.next;
        }
        console.log("\n", { end: '' });
    }
}